numpy/__init__.pyi

```diff
@@ -1651,7 +1651,7 @@ class ndarray(_ArrayOrScalarCommon, Generic[_ShapeType, _DType_co]):
     @property
     def ctypes(self) -> _ctypes[int]: ...
     @property
-    def shape(self) -> _Shape: ...
+    def shape(self) -> _ShapeType: ...
     @shape.setter
     def shape(self, value: _ShapeLike) -> None: ...
```

Comment:
This setter is a bit of a problem...

Because from typing perspective, we can't just change the shape type from e.g. `tuple[int]` to `tuple[int, int]`, even though at runtime we can

```
x = np.arange(42)
x.shape = (7, 6)
```

But if we were to only accept `value: _ShapeType` here, then `_ShapeType` cannot be covariant anymore, which kinda defeats the purpose of this entire PR.
And I'm not sure that ignoring the resulting mypy errors about invalid typevar use, is wise.

So I think that the most reasonable option is to remove this `shape` setter (only in the stubs).
Perhaps with a comment such as `The shape type is covariant and should not be written to; use .reshape() instead`.

@seberg , what do you think?

Reply:
I'm +1 on removing the stub for setting shape